test/extensions/filters/http/jwt_authn/jwt_authn_fuzz_test.cc

```diff
@@ -78,6 +78,18 @@ DEFINE_PROTO_FUZZER(const JwtAuthnFuzzInput& input) {
     return;
   }
 
+  // config.providers.value.remote_jwks.http_uri.uri gets packed into a LowerCaseString in the
+  // filter. If URI is not a valid LowerCaseString, it will crash the fuzzer. Note that the PGV
+  // well-known regex for URI is not implemented in C++, otherwise we could add a PGV rule instead
+  // of doing this check manually.
+  for (const auto& provider : input.config().providers()) {
+    absl::string_view uri = std::get<1>(provider).remote_jwks().http_uri().uri();
+    if (!Http::validHeaderString(uri)) {
```

Comment:
Is this sufficient validation for a URI?
I think it may be beneficial having a URI validator (according to the RFC), and consider using it also in the config parsing part, and reject the config if a wrong URI is provided.
WDYT?

Reply:
Makes sense. Done.

Reply:
I found [this utility](https://github.com/envoyproxy/envoy/blob/54266a082654f21b409f9d0af53a009bdd31b9de/source/common/http/utility.h#L146) for parsing URLs, but I'm not sure if it will be backward compatible. The API for this filter states the URI should be fully qualified, but take a look at the changes to tests I needed to make.